packages/@groupby/elements-search-driver-plugin/test/unit/common/search-driver-plugin.test.ts

```diff
@@ -165,11 +175,21 @@ describe('SearchDriverPlugin', () => {
         done();
       });
       sendSearchApiRequest.rejects(error);
-      searchDriverPlugin.core = {
-        [eventsPluginName]: { dispatchEvent },
-      };
+      core[eventsPluginName] = { dispatchEvent };
+
+      searchDriverPlugin.fetchSearchData({ detail: 'bad-search' } as any);
+    });
 
-      searchDriverPlugin.fetchSearchData({ detail: 'search' } as any);
+    it('should dispatch a search tracker event with origin when the search succeeds', (done) => {
+      const origin = 'some-origin';
+      response.originalResponse = { id: 'search-id' };
+      sendSearchApiRequest.resolves(response);
+      dispatchSearchBeacon.callsFake(() => {
+        expect(dispatchSearchBeacon).to.be.calledWith(response.originalResponse, origin);
+        done();
+      });
```

Comment:
This is notably a different style of test from that of the `SaytDriverPlugin`. These are more finicky, as additionally calls to the relevant fake end up triggering multiple `done()` calls. Not ideal. That being said, I don't think this is the time to refactor the rest of the file. Feel free to add any thoughts below.

Reply:
There are, but the overall testing strategy in the file is different and it would be introducing inconsistencies into this file. It's a task to be done, but I think it's tech debt.